### PR TITLE
Kris/mixpanel fix

### PR DESCRIPTION
**Pull request by Kopiczek**
Adding support for deduplication after reading the data for mixpanel

### CODE REVIEW

File: airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/export.py [hunk 1]
```diff
@@ -123,7 +123,9 @@ def process_response(self, response: requests.Response, **kwargs) -> Iterable[Ma
             if item.get("time") and item["time"].isdigit():
                 item["time"] = datetime.fromtimestamp(int(item["time"])).isoformat()
 
-            yield item
+            # Even though the API is on DATE level, we still filter items compared to state cursor
+            if not "date" in stream_state or stream_state["date"] <= item["time"]:
```
Comment (lgomezm) [inline airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/export.py:127]: are both `stream_state["date"]` and `item["time"]` strings? Should you convert them to `datetime`?

**Review by prasrvenkat — COMMENTED** [review]
Is this the API we use? https://developer.mixpanel.com/reference/raw-event-export
If so, could we use the `where` expression in query? That seems to the logical way to get incremental data.

We have start/end date which can be daily ongoing based on current cursor (_truncated to current day and previous day_) then use `mp_processing_time_ms` ([default prop](https://help.mixpanel.com/hc/en-us/articles/115004613766-Default-Properties-Collected-by-Mixpanel)) and get >= cursor value_in_ms

A quick check in our staging tells it will be truly incremental without us dropping after reading and I am sure syncs will be relatively faster too since less data every 15mins to get from export

This is what I see for past day without `where` filter
image

This is with `where` filter `properties["mp_processing_time_ms"] > 1658980000000`
image

**Comment by Kopiczek** [comment]
Yeah, I missed that option in the API. We could just use the time property,  we'll still have some duplicates but we won't require users to resync due to cursor field change

**Comment by prasrvenkat** [comment]
> Yeah, I missed that option in the API. We could just use the time property, we'll still have some duplicates but we won't require users to resync due to cursor field change

`time` is not working unfortunately, I don't know why. Maybe I was using it wrong. Few properties not working

**Pull request closed without merging**